python/kvikio/kvikio/_lib/remote_handle.pyx

```diff
@@ -23,6 +23,15 @@ cdef extern from "<kvikio/remote_handle.hpp>" nogil:
     cdef cppclass cpp_HttpEndpoint "kvikio::HttpEndpoint":
         cpp_HttpEndpoint(string url) except +
 
+    cdef cppclass cpp_S3Endpoint "kvikio::S3Endpoint":
+        cpp_S3Endpoint(string url) except +
+
+    cdef cppclass cpp_S3Endpoint "kvikio::S3Endpoint":
+        cpp_S3Endpoint(string bucket_name, string object_name) except +
```

Comment:
This is already done in my patch below, but if you don't accept it then this should be done at least.
```suggestion
    cdef cppclass cpp_S3Endpoint "kvikio::S3Endpoint":
        cpp_S3Endpoint(string url) except +
        cpp_S3Endpoint(string bucket_name, string object_name) except +
```

I would also recommend reflecting the C++ inheritance scheme into Cython like in my snippet below so that you have access to parent members if needed.